File: farmer/farmer.js
var amqp = require('amqp');

var stream = require('./lib/twitter-stream-node/lib/stream');

var config = require('./config');

function rabbitUrl() {
  if (process.env.VCAP_SERVICES) {
    conf = JSON.parse(process.env.VCAP_SERVICES);
    return conf['rabbitmq-2.4'][0].credentials.url;
  }
  else {
    return "amqp://localhost";
  }
}

function setup() {

  var exchange = conn.exchange('twitter-trends-analyze', {durable: true}, function() {

    var queue = conn.queue('', {durable: true},
    function() {
      queue.subscribe(function(msg) {
        console.log(msg);
      });
      queue.bind(exchange.name, 'tweets');
    });
    queue.on('queueBindOk', function() { gatherTweets(exchange); });
  });
}

function gatherTweets(exchange) {
	stream.public(config["username"], config["password"], function(tweet) {
	    console.log(tweet.text);
		exchange.publish('', {body: tweet.text});
	});
}

var conn = amqp.createConnection({url: rabbitUrl()});
conn.on('ready', setup);

File: farmer/farmer.rb
require 'tweetstream'
require 'json'
require 'bunny'

def amqp_url
  if (ENV["VCAP_SERVICES"])
    services = JSON.parse(ENV['VCAP_SERVICES'], :symbolize_names => true)
    url = services.values.map do |srvs|
      srvs.map do |srv|
        if srv[:label] =~ /^rabbitmq-/
          srv[:credentials][:url]
        else
          []
        end
      end
    end.flatten!.first
  else
    return "amqp://localhost"
  end
end

# Setup AMQP Messaging
b = Bunny.new
b.start
#exch = b.exchange('tweets').delete
exch = b.exchange('tweets', :type => :fanout)

# Read in the config file
configJson = File.read('config.json')
appconfig = JSON.parse(configJson)

TweetStream.configure do |config|
  config.consumer_key = appconfig["consumer_key"]
  config.consumer_secret = appconfig["consumer_secret"]
  config.oauth_token = appconfig["access_token"]
  config.oauth_token_secret = appconfig["access_secret"]
  config.auth_method = :oauth
  config.parser   = :json_pure
end

TweetStream::Client.new.sample do |status|
  puts "#{status.text}"
  # Publish the tweet to the queue
  exch.publish(status.text)
end